core/src/main/java/io/github/tcdl/Consumer.java

```diff
@@ -33,6 +35,9 @@
     private Clock clock;
 
     private MessageHandler messageHandler;
+
+    private Map<String, MessageHandler> messageHandlersByCorrelationId;
```

Comment:
Doesn't look right. We already tried to do it that way but this map doesn't make sense in case of ResponderServer.
Also we lose encapsulation that we had with CollectorManager